cfme/tests/physical_infrastructure/ui/test_physical_server_list_buttons.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+import pytest
+from selenium.common.exceptions import StaleElementReferenceException
+
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.common.physical_server_views import (
+    PhysicalServersView,
+)
+from cfme.utils.wait import wait_for
+
+pytestmark = [pytest.mark.tier(3), pytest.mark.provider([LenovoProvider], scope="module")]
+
+
+@pytest.fixture(scope="module")
+def physical_servers_collection(appliance, setup_provider_modscope):
+    # Get and return the first physical server
+    return appliance.collections.physical_servers
+
+
+@pytest.fixture(scope="module")
+def physical_servers(appliance, provider):
+    # Get and return the first physical server
+    return appliance.collections.physical_servers.all(provider)
+
+
+# Configuration Button
+def test_refresh_relationships(physical_servers_collection, provider, physical_servers):
+    last_refresh = provider.last_refresh_date()
+    physical_servers_collection.refresh(provider, physical_servers[0], physical_servers[1])
+    assert last_refresh != provider.last_refresh_date()
+
+
+actions = [
+    "power_off",
+    "power_on",
+    "power_off_now",
+    "restart",
+    "restart_now",
+    "restart_to_sys_setup",
+    "restart_to_sys_setup",
+    "blink_loc_led",
+    "turn_off_loc_led",
+    "turn_on_loc_led",
+]
+
+
+# Power / Identify Buttons
+@pytest.mark.parametrize("action", actions)
+def test_server_actions(physical_servers_collection, provider, physical_servers, action):
+    """ Test the physical server actions sending the action request, waiting the task be complete on MiQ
+        and then waiting the state of some attribute of physical server be change
+    Params:
+        * action: the action to be perform against the Physical Server
+    Metadata:
+        test_flag: crud
+    """
+    view = provider.create_view(PhysicalServersView)
+    message = 'Requested Server {} for the selected servers'.format(action)
+    getattr(physical_servers_collection, action)(physical_servers[0], physical_servers[1])
+
+    def assert_handler_displayed():
+        try:
+            if view.flash.is_displayed:
+                return view.flash.messages[0].text == message
+        except StaleElementReferenceException:
+            return False
+
+    wait_for(
+        assert_handler_displayed,
+        message="Wait for the handler alert to appear...",
+        num_sec=5,
+        delay=1
+    )
+
+
+# Lifecycle Button
+def test_lifecycle_provision(physical_server):
+    view = navigate_to(physical_server, "ProvisionCollection")
+    assert view.is_displayed
+
+
+# Policy Button
+def test_manage_button(physical_servers_collection, physical_servers):
```

Comment:
**SUGESTION:** This and the one below is only seeing that the item you are testing can be viewed, but there is no test of its functionality.   I would suggest testing the views actual functionality.

Reply:
hey @jamesooden, I created this test for only to test the buttons inside the physical server list page, I believe that the test is ok. We can create in the future a new test for this functionality, do you agree?

Reply:
I probably would have had a test for the edit functionality that would have tested the button as part of its testing.   My comment was a suggestion though, and certainly we can add more testing later.

Reply:
@CharlleDaniel Yes, we can always grow the tests in the future, or add new ones, but this effectively boils down to a unit test of the widget for the dropdown, and of the is_displayed implementation for the given view.

We're much more interested in functional testing of the features - during which we will be asserting views are displayed, and widgets interacted with.  Even a simple CRUD test would be preferred, where a policy is created, assigned to the resource, we validate the policy assignment, update if relevant, and then remove the policy assignment.  Not going more in depth (asserting enforcement of the policy) keeps the test content simple, but goes a lot further to validate simple operation of the policies feature with respect to physical servers, and will verify several views/widgets along the way. (collection, details, policies form, dropdowns, flash messages, collections table/grid, etc)